scripts/start_port.sh

```diff
@@ -68,7 +68,7 @@ echo "Applying automated fixes"
     sed -i '/^import/{s/[.]Gcd/.GCD/g; s/[.]Modeq/.ModEq/g; s/[.]Nary/.NAry/g; s/[.]Peq/.PEq/g; s/[.]Pfun/.PFun/g; s/[.]Pnat/.PNat/g; s/[.]Smul/.SMul/g; s/[.]Zmod/.ZMod/g}' "$mathlib4_path"
 
     # awk script taken from https://github.com/leanprover-community/mathlib4/pull/1523
-    awk '{do {{if (match($0, "^  by$") && length(p) < 98 && (!(match(p, "^[ \t]*--.*$")))) {p=p " by";} else {if (NR!=1) {print p}; p=$0}}} while (getline == 1) if (getline==0) print p}' "$mathlib4_path" > "$mathlib4_path.tmp"
+    awk '{do {{if (match($0, "^  by$") && length(p) < 98 && (!(match(p, "^[ \t]*--.*$")))) {p=p " by";} else {if (NR!=1) {print p}; p=$0}}} while (getline == 1); if (getline==0) print p}' "$mathlib4_path" > "$mathlib4_path.tmp"
```

Comment:
```suggestion
    # UTF8-encoded checkmark
    printf "\xE2\x9C\x94" | awk "{exit length != 1}" || exit "Your version of awk does not handle unicode correctly; consider installing \`gawk\`"
    awk '{do {{if (match($0, "^  by$") && length(p) < 98 && (!(match(p, "^[ \t]*--.*$")))) {p=p " by";} else {if (NR!=1) {print p}; p=$0}}} while (getline == 1); if (getline==0) print p}' "$mathlib4_path" > "$mathlib4_path.tmp"
```

Reply:
note that
1) you can directly call `gawk` with the `gawk` command always, so this is easier
2) this can fail with `gawk` if `LC_ALL=C`, or any other ASCII-based locale

Reply:
anyway, I will solve this by shelling out to `gawk` exactly, and specifying locale in the next version of the patch.

Reply:
Also, I think we should perform the check that `gawk` exists at the top of the file, to prevent partial script runs

Reply:
Partial script runs aren't really a problem, the run happens in a temporary directory.